tricycle_controller/src/tricycle_controller.cpp

```diff
@@ -637,31 +637,29 @@ CallbackReturn TricycleController::get_steering(
   return CallbackReturn::SUCCESS;
 }
 
-double TricycleController::convert_trans_rot_vel_to_steering_angle(
-  double Vx, double theta_dot, double wheelbase)
-{
-  if (theta_dot == 0 || Vx == 0)
-  {
-    return 0;
-  }
-  return std::atan(theta_dot * wheelbase / Vx);
-}
-
-std::tuple<double, double> TricycleController::twist_to_ackermann(double Vx, double theta_dot)
+// double TricycleController::convert_trans_rot_vel_to_steering_angle(
+//   double Vx, double theta_dot, double wheelbase)
+// {
+//   if (theta_dot == 0 || Vx == 0)
+//   {
+//     return 0;
+//   }
+//   return std::atan(theta_dot * wheelbase / Vx);
+// }
+
+std::tuple<double, double> TricycleController::twist_to_tricycle(double linear_command, double angular_command)
 {
   // using naming convention in http://users.isr.ist.utl.pt/~mir/cadeiras/robmovel/Kinematics.pdf
-  double alpha, Ws;
-
-  if (Vx == 0 && theta_dot != 0)
-  {  // is spin action
-    alpha = theta_dot > 0 ? M_PI_2 : -M_PI_2;
-    Ws = abs(theta_dot) * wheel_params_.wheelbase / wheel_params_.radius;
-    return std::make_tuple(alpha, Ws);
-  }
+  double traction_velocity = sqrt(pow((angular_command * wheel_params_.wheelbase),2) + pow((linear_command),2)) / wheel_params_.radius;
```

Comment:
in comment, there's a reference to the nomenclature used in the "old math"

please do something similar for the "new math" so we don't need to reverse-compile code to math

Reply:
@ros2torial any reason why you want to change the nomenclature in the first place? Is it not clear enough with the documentation provided?

Reply:
@bmagyar @tonynajjar I have kept **linear_command** and **angular_command** name as it is easy to understand and same nomenclature is used in diff_drive_controller. Regarding alpha, I have changed its name to **steering_wheel_angle** in the latest commit as it is also easy to understand and this name is as per the nomenclature only (_If the **steering wheel** is set to an **angle** α(t) from the straight-line direction_).  
Let me know if the new variable names are ok else we can keep the previous names.